torchseg/encoders/timm.py

```diff
@@ -0,0 +1,144 @@
+import timm
+import torch.nn as nn
+from einops import rearrange
+
+from .supported import TIMM_ENCODERS
+
+
+class TimmEncoder(nn.Module):
+    def __init__(
+        self,
+        name,
+        pretrained=True,
+        in_channels=3,
+        depth=None,
+        indices=None,
+        output_stride=32,
+        **kwargs,
+    ):
+        super().__init__()
+
+        assert (
+            depth is not None or indices is not None
+        ), "Either `depth` or `indices` should be specified"
+
+        if indices is not None:
+            depth = len(indices)
+        else:
+            indices = tuple(range(depth))
+
+        total_depth = TIMM_ENCODERS[name.split(".")[0]]["indices"]
+        if len(total_depth) < depth:
+            err = f"""
+            The specified depth={depth}or indices={indices} is greater than
+            the maximum available depth={total_depth} for the {name} encoder.
+            """
+            raise ValueError(err)
+
+        params = dict(
+            in_chans=in_channels,
+            features_only=True,
+            pretrained=pretrained,
+            out_indices=indices,
+            output_stride=output_stride,
+        )
```

Comment:
Maybe we can add the custom `norm_layer` and `act_layer` directly here in `params` (and in `__init__`) if specified instead of leaving them in kwargs, just to make it clearer, what do you think? I just don't know how pretrained weights are handled in case of different norm and activation layers from the usual `BatchNorm2d` and `ReLU`, maybe we could force `pretrained` to False for now